lib/addHooks.js

```diff
@@ -42,6 +42,11 @@ module.exports = function addHooks(runner, transactions, callback) {
     runner.hooks.transactions[transaction.name] = transaction;
   });
 
+  // No hooks
+  if (!runner.configuration.hookfiles) {
```

Comment:
You are right, this branch doesn't cover the scenario when there are no hooks. `unner.configuration.hookfiles` is an empty array `[]`, which resolves to `false` in this condition.

The code follow down, and after:

```js
let hookfiles;
  try {
    hookfiles = resolvePaths(runner.configuration.custom.cwd, runner.configuration.hookfiles);
  } catch (err) {
    return callback(err);
  }

// Override hookfiles option in configuration object with
// sorted and resolved files
runner.configuration.hookfiles = hookfiles;
``` 

After this you mutate the `configuration.hookfiles` with `hookfiles`. `resolvePaths` returns an empty array in case there were no hooks (empty array).

Probably, you could assert the length of hooks:

```js
if (runner.configuration.hookfiles && runner.configuration.hookfiles.length ===0) {
  return callback(err)
}
```

Reply:
Yes, fixing the code is easy. Fixing the test to actually test something will be tricky.